src/Mod/TechDraw/Gui/TaskDimension.cpp

```diff
@@ -124,6 +146,17 @@ TaskDimension::TaskDimension(QGIViewDimension *parent, ViewProviderDimension *di
     connect(ui->cbArbitraryTolerances, &QCheckBox::stateChanged, this, &TaskDimension::onArbitraryTolerancesChanged);
 #endif
 
+    // Reference
+    std::regex refRegex("\\(%\\.([0-9]+)([fFrRgGwWeE])\\)");
+    if (std::regex_search(currentFormat, refRegex)) {
+        ui->cbReference->setChecked(true);
+    } else {
+        ui->cbReference->setChecked(false);
+    }
```

Comment:
```suggestion
    const bool hasReference = std::regex_search(currentFormat, refRegex);
    ui->cbReference->setChecked(hasReference);
```

Something like that would be easier to read. Name for the variable should be better probably.